mrpt_graphslam_2d/CMakeLists.txt

```diff
@@ -18,3 +18,49 @@ if (BUILD_TESTING)
     ${test_output_directory}/${PROJECT_TEST_NAME})
 endif()
 
+set(PROJECT_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../)
+
+#V-REP Config
+################################################################################
+
+if (EXISTS "${PROJECT_ROOT_DIR}/../V-REP")
+  message(STATUS "Found V-REP")
+  set (V-REP_ROOT "${PROJECT_ROOT_DIR}/../" CACHE "V-REP root dir" STRING)
+
+  execute_process (COMMAND "/bin/bash" "-c" "cd ${V-REP_ROOT}; echo -n `pwd`" OUTPUT_VARIABLE V-REP_ROOT)
+
+
+  set (V-REP_PATH "${PROJECT_ROOT_DIR}/../V-REP/programming/" CACHE "The V-REP build path" STRING)
+
+  set (SOURCES remoteApi/bubbleRob_test.cpp
+    ${V-REP_ROOT}/V-REP/programming/remoteApi/extApi.c
+    ${V-REP_ROOT}/V-REP/programming/common/shared_memory.c
+    ${V-REP_ROOT}/V-REP/programming/remoteApi/extApi.h
+    ${V-REP_ROOT}/V-REP/programming/remoteApi/extApiPlatform.h
+    ${V-REP_ROOT}/V-REP/programming/include/shared_memory.h
+    ${V-REP_ROOT}/V-REP/programming/remoteApi/extApiPlatform.c)
+
+  set (EXTRA_LIBRARIES pthread)
+
+  include_directories (${V-REP_PATH}/remoteApi ${V-REP_PATH}/include)
+  
+  add_definitions( -Dpthread )
```

Comment:
@bergercookie I don't know how much support you'll be planning to add for building in Windows, but at least I think we should avoid directly assuming we're in Linux & gcc, don't you agree?

Reply:
Hmm, I haven't run vrep in Windows so I don't know how to handle it for now.
@jlblancoc I suggest we first get a working version running and then add some extra commits to address releasing for windows... Obviously though  in places (`execute_process(bash "-c", ...)`) we should avoid it in the first place